ERCS/erc-9999.md

```diff
@@ -0,0 +1,134 @@
+     * @param vTokenAmount Amount of vTOKEN.
+     * @return uint256 Equivalent native token amount.
+     */
+    function calculateNativeEquivalent(uint256 vTokenAmount) external view returns (uint256);
+
+    /**
+     * @dev Returns the address of the paired native token.
+     * @return address Native token contract.
+     */
+    function nativeToken() external view returns (address);
+}
+```
+
+### Behavior
+
+* Transfers to EOAs **MUST** trigger a conversion: vTOKENs are burned from the sender, and the calculated native token equivalent is transferred to the EOA.
+* Transfers between whitelisted contracts **MUST** proceed as standard ERC-20 transfers.
+* Any transfer attempt to an unauthorized recipient address (i.e., not an EOA triggering conversion, and not a whitelisted contract) **MUST** cause a revert.
+
+## Rationale
+
+* **Transfer Override**: Overriding the standard `_transfer` function **MUST** be implemented to ensure all transfer logic paths correctly enforce the virtual conversion and whitelist rules. This maintains ERC-20 compatibility while adding the unique vTOKEN behavior.
+* **Whitelist Enforcement**: An explicit whitelist **SHOULD** be used to prevent unauthorized contract interactions and allow precise control over vTOKEN circulation. Typical whitelisted addresses include liquidity pool contracts, routers, and protocol treasuries.
+* **Burn-on-convert Model**: vTOKENs **MUST** be burned during conversion to prevent supply inflation and ensure the conservation of value. This mechanism maintains accurate supply metrics.
+* **Deterministic Conversion Logic**: The `calculateNativeEquivalent` function **SHOULD** implement deterministic or oracle-based conversion rates to avoid manipulation. The rate **MAY** be fixed (e.g., 1:1) or dynamically derived (e.g., from an AMM or TWAP oracle).
+
+## Backwards Compatibility
+
+ERC-vTOKEN maintains:
+
+* Compatibility with existing [ERC-20](https://www.google.com/search?q=./eip-20.md) tools and functions, such as `balanceOf`, `allowance`, and `transferFrom` (when interacting with whitelisted contracts).
+* Consistency in event emissions and metadata (e.g., `name()`, `symbol()`, `decimals()`).
+
+It is **not compatible** with:
+
+* Standard cryptocurrency wallets that expect to hold and directly transfer tokens from an EOA.
+* Decentralized Exchange (DEX) aggregators or other protocols that do not explicitly integrate or understand the vTOKEN conversion mechanism.
+
+## Test Cases
+
+Comprehensive test cases **SHOULD** be provided in the reference repository to cover critical scenarios, including:
+
+* Transfer to an EOA **MUST** trigger conversion to native tokens.
+* Transfer to a whitelisted contract **MUST** be allowed and behave as a standard ERC-20 transfer.
+* Transfer to an unknown (non-whitelisted, non-EOA) address **MUST** revert.
+* Conversion with insufficient native token balance in the vTOKEN contract **MUST** revert.
```

Comment:
Test cases should be simple inputs/outputs/state changes, or possibly unit tests defined with some framework. They cannot just be vague descriptions of what test cases will eventually exist.

Note that you do not have to provide test cases. For now, I'd just remove this section and add it back later if you decide to spec them out.